src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -6190,7 +6190,11 @@ static QVariant fcnArrayInsert( const QVariantList &values, const QgsExpressionC
 static QVariant fcnArrayRemoveAt( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
 {
   QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
-  list.removeAt( QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent ) );
+  int position = QgsExpressionUtils::getNativeIntValue( values.at( 1 ), parent );
+  if ( position < 0 && ( list.length() + position ) >= 0 )
```

Comment:
```suggestion
  if ( position < 0 )
```
the extra check is redundant, since it's being performed on line 6196